src/block/Sugarcane.php

```diff
@@ -122,14 +122,25 @@ public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Blo
 		$down = $this->getSide(Facing::DOWN);
 		if($down->isSameType($this)){
 			return parent::place($tx, $item, $blockReplace, $blockClicked, $face, $clickVector, $player);
-		}elseif($down->getId() === BlockLegacyIds::GRASS || $down->getId() === BlockLegacyIds::DIRT || $down->getId() === BlockLegacyIds::SAND || $down->getId() === BlockLegacyIds::PODZOL){
+		}elseif($this->isValidSupport($down)){
 			foreach(Facing::HORIZONTAL as $side){
-				if($down->getSide($side) instanceof Water){
+				if($down->getSide($side) instanceof Water || $down->getSide($side) instanceof FrostedIce){
 					return parent::place($tx, $item, $blockReplace, $blockClicked, $face, $clickVector, $player);
 				}
 			}
 		}
 
 		return false;
 	}
+
+	private function isValidSupport(Block $block) : bool{
+		$id = $block->getId();
+		//TODO: rooted dirt, moss block
+		return $id === BlockLegacyIds::SUGARCANE_BLOCK
```

Comment:
`isSameType(this)` would be better than hardcoding SUGARCANE_BLOCK; plugins may want to create custom sugarcane blocks